### ISSUE TITLE
KopiaUI: Unable to disconnect repository where storage target is unavailable

### ISSUE THREAD

**Issue by typpos** [kopia/kopia#2452]
I'm unable to disconnect a repository where the target storage medium is unavailable.

Version v0.11.3 317cc36892707ab9bdc5f6e4dea567d1e638a070 kopia/htmlui fcd217241a877fdba53a3afa72d68b73f14fd8c2 built on Thu Jul 7 03:07:56 UTC 2022 fv-az196-146

Windows 10 Pro x64, 10.0.19043

Reproduce:
* Start KopiaUI
* Create a new repository, "Test", on a removable drive or a remote network share
* Create a policy and perform a backup
* Close the KopiaUI Window
* Remove the storage device or disconnect the network share
* Windows Task Bar > KopiaUI Icon > [R-Click] > Click on "test" 
* KopiaUI shows empty window with title bar "KopiaUI is loading...".
* Issue: This window never completes or allows cancelation or "disconnect"
* If you close the window, a background task remains active and consumes memory and sometimes resources

How do I disconnect and remove the repository?

**Comment by dimejo** [comment]
Can you try renaming/removing the configuration file? On Windows it should be located under `%APPDATA%\kopia`.

**Comment by typpos** [comment]
Thanks. That workaround works.

**Comment by LinuxOnTheDesktop** [comment]
This problem is one of the (too many!) points that I made in 2419. Ideally, one should not need a workaround - but I am glad to learn that there is one. 

Actually: we really need more than a workaround. For, enacting the workaround seemed hard. I presume I was to do the following: close the Kopia UI and delete a file within `%APPDATA%\kopia`? But _which_ file to delete? For I have a working repository and an unworking one and I want to remove only the unworking one. I can inspect the two config (.config) files in a text editor but, since the working repository was a new version of the old one, many of the lines in the two files are identical. Also, each repository perhaps has _two_ files (a .config file and a .config.mlock file); I tried playing around with the files - moving to another directory - now one group and now another - but I could not get things working as they should.

**Comment by dimejo** [comment]
> But which file to delete? For I have a working repository and an unworking one and I want to remove only the unworking one.

On most tabs in the KopiaUI you will see a yellow button at the bottom of the output. Click on it and Kopia will show you the command and configuration file it was using. If you only have 2 repositories it should be easy to distinguish the 'good' and the 'bad' repository.

**Comment by LinuxOnTheDesktop** [comment]
Thanks, but problems persist. Let me explain.

By using the `description` field and the config-file-shown-in-terminal, I discovered which config file was the problematic one. Then I closed the kopia UI and removed the file. I removed the associated 'mlock' file too. Then I restarted Kopia. Kopia showed a window asking me to create a new repository and _the tray icon still shows both my good repository and an entry that reads 'not connected'_.

**Comment by LinuxOnTheDesktop** [comment]
I think that there is some other configuration file that I need to delete or edit. Yet, Kopia has a lot of configuration files . . Also: when rooting around, I found one log file that was 40MB (which seems too big, no?).

**Comment by LinuxOnTheDesktop** [comment]
I have discovered the following sort-of-workaround.

When I start Kopia, it prompts me to configure the persisting-can't-get-rid-of-it semi-or-brokenly configured repository. If I _do_ create a repository, using local storage, and call it 'DUMMY', then:

1) on Kopia's tray icon, 'unconnected' gets replaced by 'DUMMY';
2) when I start Kopia, I no longer see an irritating notification about Kopia running in the background;
3) when I start Kopia, it no longer asks me to configure a repository.

So that is the workaround.

If I 'disconnect' the 'DUMMY' repository, then the behaviour of 1-3 reverts.

_Is it that Kopia needs a file to be called `repository.config`?_ I tried deleting the `repository.config` and `repository.config.mlock` - both those files having been created by the dummy repository - and then renaming the files for my good repository such that they lacked numbers (i.e. `repository-<numbers>.config` became `repository.config`, and `repository-<numbers>.config.mlock` became `repository.config.mlock`). The result was that things got borked even further. So I reverted those changes.
